<?xml version="1.0"?>
<launch>
    <arg name="map_2d" default="false" />
    <arg name="bag_file" default="/home/share/rosbag/2022-07-06-02-07-20.bag" />
    <arg name="playback_speed" default="3.0" />

    <param name="/use_sim_time" value="true" />

    <group if="$(arg map_2d)">
        <node pkg="gmapping" type="slam_gmapping" name="mapper">
		<param name="maxUrange" value="80.0" type="double" />
		<param name="xmax" value="30" type="double" />
		<param name="ymax" value="30" type="double" />
		<param name="xmin" value="-30" type="double" />
		<param name="ymin" value="-30" type="double" />
	    </node>
        <node name="teensy_handler" pkg="kcctcore" type="teensy_handler"/>
        <include file="$(find kcctcore)/launch/sensor/velodyne/pc2scan.launch"/>
        <node pkg="kcctcore" type="launch_delayed.sh" args="0 rosbag play $(arg bag_file) -r $(arg playback_speed) --clock --topic /velodyne_points /float_sensor_data" name="launch_delayed1" output="screen" />
    </group>
    <group unless="$(arg map_2d)">
        <include file="$(find lego_loam)/launch/run.launch"/>
        <node pkg="kcctcore" type="launch_delayed.sh" args="0 rosbag play $(arg bag_file) -r $(arg playback_speed) --clock --topic /velodyne_points" name="launch_delayed1" output="screen" />
    </group>
    
    
</launch>
